### PR TITLE
Imperative transfer gru unit

### PR DESCRIPTION
**Pull request by velconia**
1. transfer gru unit layer
2. support imperative check in op test's check_output
3. Add Layers UT for gru

### CODE REVIEW

File: python/paddle/fluid/imperative/nn.py [hunk 1]
```diff
@@ -468,3 +469,136 @@ def forward(self, input):
...
+    connected layer has to be applied, such that :math:`z_t = W_{fc}x_t`.
+
+    The terms :math:`u_t` and :math:`r_t` represent the update and reset gates
+    of the GRU cell. Unlike LSTM, GRU has one lesser gate. However, there is
+    an intermediate candidate hidden output, which is denoted by :math:`m_t`.
+    This layer has three outputs :math:`h_t`, :math:`dot(r_t, h_{t-1})`
+    and concatenation of :math:`u_t`, :math:`r_t` and :math:`m_t`.
+
+    Args:
+        input (Variable): The fc transformed input value of current step.
```
Comment (panyx0718) [inline python/paddle/fluid/imperative/nn.py:518]: add name_scope?

File: python/paddle/fluid/imperative/nn.py [hunk 2]
```diff
@@ -468,3 +469,136 @@ def forward(self, input):
...
+            sigmoid=1,
+            tanh=2,
+            relu=3, )
+        activation = activation_dict[activation]
+        gate_activation = activation_dict[gate_activation]
+
+        self._dtype = dtype
+        size = size // 3
+        # create weight
+        self._weight = self.create_parameter(
```
Comment (panyx0718) [inline python/paddle/fluid/imperative/nn.py:574]: I'm thinking. maybe we should make parameter public so people can use it.

**Review by panyx0718 — DISMISSED** [review]

File: python/paddle/fluid/imperative/nn.py (hunk 1 above)
Comment (velconia) [reply python/paddle/fluid/imperative/nn.py:518]: OK, I'll add it in next PR

File: python/paddle/fluid/imperative/nn.py (hunk 2 above)
Comment (velconia) [reply python/paddle/fluid/imperative/nn.py:574]: Can't agree more~

File: python/paddle/fluid/imperative/nn.py (hunk 1 above)
Comment (velconia) [reply python/paddle/fluid/imperative/nn.py:518]: Done

**Review by panyx0718 — APPROVED** [review]

**Pull request merged**